### PR TITLE
Remove the wrong configuration from prowgen configs and regenerate jobs

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/test-platform 

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by petr-muller** [comment]
/hold

Why would intentional configuration be wrong?

**Comment by droslean** [comment]
> Why would intentional configuration be wrong?
Those repos are not private, and we unnecessarily generate the jobs passing an OAuth authentication + the secrets. Also, any new changes that affect only the private jobs will end up breaking the jobs in this PR.

File: ci-operator/config/open-cluster-management/api/.config.prowgen [hunk 1]
```diff
@@ -1,2 +1 @@
-private: true
```
Comment (petr-muller) [inline ci-operator/config/open-cluster-management/api/.config.prowgen:1]: We can delete the files entirely. `expose: true` is only effective with `private: true`

**Comment by petr-muller** [comment]
/hold cancel

Looks like all the repos are indeed public. It'd be good to let someone from OCM ack this PR. @ray-harris maybe?

**Comment by ray-harris** [comment]
The reason these have the `.config/prowgen` was not that they were private, but they originally cloned private repos during their build. We had also added the code for the `exposed: true` field so that we could still have the prow status visible.

I don't think they rely on private repos any more, however, so I'm OK with this change.

/lgtm

**Comment by droslean** [comment]
@petr-muller Since @ray-harris approved, we will need you to tag the PR as well.

**Comment by droslean** [comment]
@ray-harris I had to rebase. Can you re-lgtm?

**Pull request merged**